### PR TITLE
nethack: update to 3.6.1, keep 3.4.3 as coexisting subport

### PR DESCRIPTION
**Pull request by jflude**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.14.3 18D109
Xcode 10.1 10B61 

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by mf2k** [comment]
The sub-port only needs to add fields that are different than the main port. So most of the lines in it can be deleted. Yes, I know we need to document it better. You can look at other ports with sub-ports for now. As well as comments from Macports committers like me. :)

File: games/nethack/Portfile [hunk 1]
```diff
@@ -1,110 +1,200 @@
...
 }
 
-variant autopickup_exceptions description { Automatically pick up things onto which you move } {
-    patchfiles-append patch-include__config.h.diff
-}
+subport nethack343 {
+    name                nethack343
+    version             3.4.3
+    revision            4
+    categories          games
```
Comment (mf2k) [inline games/nethack/Portfile]: This line can be removed because it comes from the main port.

File: games/nethack/Portfile [hunk 2]
```diff
@@ -1,110 +1,200 @@
...
     depends_lib-append port:xorg-libXaw
 }
 
-variant autopickup_exceptions description { Automatically pick up things onto which you move } {
-    patchfiles-append patch-include__config.h.diff
-}
+subport nethack343 {
+    name                nethack343
+    version             3.4.3
+    revision            4
```
Comment (mf2k) [inline games/nethack/Portfile]: The revision should be set to 0.

File: games/nethack/Portfile [hunk 3]
```diff
@@ -1,110 +1,200 @@
...
+    categories          games
+    maintainers         @Nax
```
Comment (mf2k) [inline games/nethack/Portfile]: This line can be deleted too.

File: games/nethack/Portfile [hunk 4]
```diff
@@ -1,110 +1,200 @@
...
+    homepage            http://nethack.sourceforge.net/
+    platforms           darwin freebsd
+    license             Copyleft
+
+    master_sites        sourceforge:nethack
+    distname            nethack-343-src
+    extract.suffix      .tgz
+    checksums           ${distfiles} md5 21479c95990eefe7650df582426457f9 \
+                        ${distfiles} sha1 c26537093c38152bc0fbcec20468d975b35f59fd \
+                        ${distfiles} rmd160 42f600d24715a0b7e631b357c135761410b3ca95
```
Comment (mf2k) [inline games/nethack/Portfile]: The checksums can remain because it is a different file. But remove the references to` ${distfiles}`.

Comment (jflude) [reply games/nethack/Portfile]: Yes, I wondered about that - I think the `${distfiles}` is a hangover from when the commented out variants included checksums-append.

File: games/nethack/Portfile (hunk 2 above)
Comment (jflude) [reply games/nethack/Portfile]: lines 89, 91 deleted

File: games/nethack/Portfile (hunk 3 above)
Comment (jflude) [reply games/nethack/Portfile]: lines 92, 93 deleted

File: games/nethack/Portfile (hunk 4 above)
Comment (jflude) [reply games/nethack/Portfile]: line 102 deleted; removed `${distfiles}` from checksums

File: games/nethack/Portfile (hunk 1 above)
Comment (jflude) [reply games/nethack/Portfile]: done

**Comment by jflude** [comment]
Hi @mf2k - all changes completed, awaiting yours and @nax's approval - thanks!

**Comment by jflude** [comment]
Hold off approving this, please, it also builds against XQuartz and not MacPorts Xorg - fix coming...

**Comment by jflude** [comment]
XQuartz issue fixed, back to awaiting approval...

**Comment by jflude** [comment]
Looks like it would have passed lint, except: `Error: Portfile is from the future - check date and time of your system`

**Comment by pmetzger** [comment]
@mf2k ?

File: games/nethack/Portfile [hunk 5]
```diff
@@ -1,110 +1,190 @@
...
 }
 
-variant autopickup_exceptions description { Automatically pick up things onto which you move } {
-    patchfiles-append patch-include__config.h.diff
-}
+subport nethack343 {
+    version             3.4.3
+    description         Classic dungeon adventure game (previous version).
+    homepage            http://nethack.sourceforge.net/
+    platforms           darwin freebsd
```
[the remark this answers is not in the document]
Comment (jflude) [reply games/nethack/Portfile]: Sorry, I'm not sure I understand - all 4 lines above are different in the 343 sub-port and the 361 main-port.  Can you be more specific, giving the line numbers of the lines you think should be changed?

File: games/nethack/Portfile [hunk 6]
```diff
@@ -1,110 +1,190 @@
...
     depends_lib-append port:xorg-libXaw
 }
 
-variant autopickup_exceptions description { Automatically pick up things onto which you move } {
-    patchfiles-append patch-include__config.h.diff
-}
+subport nethack343 {
+    version             3.4.3
+    description         Classic dungeon adventure game (previous version).
+    homepage            http://nethack.sourceforge.net/
```
[the remark this answers is not in the document]
Comment (jflude) [reply games/nethack/Portfile]: Do you mean line 89?

[the remark this answers is not in the document]
Comment (mf2k) [reply games/nethack/Portfile]: Line 91. The `homepage`. Please scan for any such duplicate lines in the nethack343 sub-port.

File: games/nethack/Portfile [hunk 7]
```diff
@@ -1,110 +1,190 @@
...
+    configure.cmd       /bin/sh
+    configure.pre_args  setup.sh
+
+    build.args-append   CC=${configure.cc} \
+                        CXX=${configure.cxx} \
+                        CPP=${configure.cpp} \
+                        CFLAGS="-O2 -I../include"
+
+    pre-destroot {
+        file mkdir "${destroot}${prefix}/share/man/man6"
```
[the remark this answers is not in the document]
Comment (mf2k) [reply games/nethack/Portfile]: Line 124 did not get updated.

File: games/nethack/Portfile [hunk 8]
```diff
@@ -1,110 +1,190 @@
...
+        file mkdir "${destroot}${prefix}/share/nethackdir/save"
```
[the remark this answers is not in the document]
Comment (mf2k) [reply games/nethack/Portfile]: And Line 125.

File: games/nethack/Portfile (hunk 7 above)
[the remark this answers is not in the document]
Comment (jflude) [reply games/nethack/Portfile]: Sure, but I took the approach of taking the old 343 main-port and making it a sub-port of 361 with the minimum of changes, to reduce the risk of breaking anything.  Happy to tidy it up if you think it's worth it.

File: games/nethack/Portfile (hunk 8 above)
[the remark this answers is not in the document]
Comment (jflude) [reply games/nethack/Portfile]: Same comment as above.

File: games/nethack/Portfile [hunk 9]
```diff
@@ -1,110 +1,190 @@
...
+        }
+    }
+
+    variant x11 {
+        patchfiles-append \
+            x11/patch-include__config.h.diff \
+            x11/patch-sys__unix__Makefile.src.diff \
+            x11/patch-sys__unix__Makefile.top.diff
+        depends_lib-append port:xorg-libXaw
+    }
```
[the remark this answers is not in the document]
Comment (mf2k) [reply games/nethack/Portfile]: This did not get changed.

[the remark this answers is not in the document]
Comment (jflude) [reply games/nethack/Portfile]: Will do.  (In fact, the revised code would need to be

File: games/nethack/Portfile (hunk 6 above)
[the remark this answers is not in the document]
Comment (jflude) [reply games/nethack/Portfile]: The homepage is www.nethack.org for 361 and nethack.sourceforge.net for 343 (as per the original port).  It certainly makes sense to change the homepage for 343 (sourceforge was not the right page for 343 either, IMHO) - but I want to make sure - you are definitely looking at the latest code, right?  As this it not a duplicate line...

File: games/nethack/Portfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (jflude) [reply games/nethack/Portfile]: Removed `homepage` (line 91) from sub-port.

File: games/nethack/Portfile (hunk 6 above)
[the remark this answers is not in the document]
Comment (jflude) [reply games/nethack/Portfile]: Done

File: games/nethack/Portfile (hunk 7 above)
[the remark this answers is not in the document]
Comment (jflude) [reply games/nethack/Portfile]: Combined `pre-destroot` for both main-port and sub-port (not wild about that, I think Portfiles would be clearer with less conditional logic even if there is more redundancy, but happy to follow the guidance)

File: games/nethack/Portfile (hunk 8 above)
[the remark this answers is not in the document]
Comment (jflude) [reply games/nethack/Portfile]: Done

File: games/nethack/Portfile (hunk 9 above)
[the remark this answers is not in the document]
Comment (jflude) [reply games/nethack/Portfile]: Done

**Comment by jflude** [comment]
All suggested changes implemented...

File: games/nethack/Portfile (hunk 4 above)
Comment (mf2k) [reply games/nethack/Portfile]: I am still seeing the `${distfiles}` here. Also we normally have a `sha256` checksum and NOT `md5` and `sha1`. Something like this:

```
checksums           rmd160 42f600d24715a0b7e631b357c135761410b3ca95 \
                    sha256 .... \
                    size   ...

```

Comment (jflude) [reply games/nethack/Portfile]: Done

**Comment by mf2k** [comment]
I think more could be done to reduce duplication. Maybe this port can be made `openmaintainer` so that I can make the changes?

**Comment by jflude** [comment]
Thanks, Frank.  Yes, I think at least there could be common code to preserve/recreate the game state files, shared by the main-port and sub-port.  I can make a quick change to do that and submit a new PR.  Very happy to add `openmaintainer` as well, although I guess that is @Nax's call...?

**Comment by Nax** [comment]
Sure, go ahead.

**Comment by mf2k** [comment]
Yes it is, but please add `openmaintainer` to a new PR.

**Pull request merged**